src/reducers/ui.js

```diff
@@ -51,11 +74,13 @@ export default function ui(stateIn, action) {
       return state.set('workspace', DEFAULT_WORKSPACE);
 
     case 'HIDE_COMPONENT':
+      return restoreFlexOnComponentToggle(action.payload.componentName, state);
+
     case 'UNHIDE_COMPONENT':
-      if (action.payload.componentName === 'output') {
-        return state.setIn(['workspace', 'rowFlex'], DEFAULT_ROW_FLEX);
-      }
-      return state.setIn(['workspace', 'columnFlex'], DEFAULT_COLUMN_FLEX);
+      state = focusEditor(
+        action.payload.componentName.replace('editor.', ''),
+        state);
```

Comment:
Very minor nitpick, but project style would be to put the `)` on the next line : )